pippo-core/src/main/java/ro/fortsoft/pippo/core/Response.java

```diff
@@ -576,7 +576,7 @@ public void send(CharSequence content) {
     public void resource(InputStream input) {
         checkCommitted();
 
-        if (isHeaderEmpty(HttpConstants.Header.CONTENT_TYPE)) {
+        if (isHeaderEmpty(HttpConstants.Header.CONTENT_TYPE) && getHttpServletResponse().getContentType() == null) {
             header(HttpConstants.Header.CONTENT_TYPE, HttpConstants.ContentType.APPLICATION_OCTET_STREAM);
         }
 
```

Comment:
We should not even look at the header for content type.
```java
if (getContentType() == null) {
    contentType(HttpConstants.ContentType.APPLICATION_OCTET_STREAM);
}
```

should also change the `commit()` method to use the same approach.